plugins/modules/postgresql_set.py

```diff
@@ -240,33 +240,57 @@ def pretty_to_bytes(pretty_val):
     # if the value contains 'B', 'kB', 'MB', 'GB', 'TB'.
     # Otherwise it returns the passed argument.
 
+    # It's sometimes possible to have an empty values
+    if not pretty_val:
+        return pretty_val
+
+    # If the first char is not a digit, it does not make sense
+    # to parse further, so just return a passed value
+    if not pretty_val[0].isdigit():
+        return pretty_val
+
+    # If the last char is not an alphabetical symbol, it means that
+    # it does not contain any suffixes, so no sense to parse further
+    if not pretty_val[-1].isalpha():
+        return pretty_val
+
+    # Extract digits
+    num_part = []
+    for c in pretty_val:
+        # When we reach the first non-digit element,
+        # e.g. in 1024kB, stop iterating
+        if not c.isdigit():
+            break
+
+        if c.isdigit():
+            num_part.append(c)
+
+    num_part = ''.join(num_part)
+
     val_in_bytes = None
 
-    if 'kB' in pretty_val:
-        num_part = int(''.join(d for d in pretty_val if d.isdigit()))
-        val_in_bytes = num_part * 1024
+    if len(pretty_val) >= 2:
+        if 'kB' in pretty_val[-2:]:
+            val_in_bytes = num_part * 1024
 
-    elif 'MB' in pretty_val.upper():
-        num_part = int(''.join(d for d in pretty_val if d.isdigit()))
-        val_in_bytes = num_part * 1024 * 1024
+        elif 'MB' in pretty_val[-2:]:
+            val_in_bytes = num_part * 1024 * 1024
```

Comment:
What about `mb`? Is this still covered like in line 249 with `pretty_val.upper()`?
Same is for `GB`, `TB` and `B`, please check for case sensitive.

Reply:
@tcraxs yes, it should be covered, i consider this `.upper()` is extra because of line 345 in `main()`:
```
        for unit in POSSIBLE_SIZE_UNITS:
            if (value[:-2].isdigit() and unit in value[-2:]) or ('b' in value[-1] and value[:-1].isdigit()):
                value = value.upper()
```
and the function is called after this transformation so removed.
Also the CI tests cover the case you mentioned

Reply:
Okay, perfect. I just oversaw this (also in the test cases).

Reply:
np, I also oversaw this when writing the module;)